flocker/node/test/test_deploy.py

```diff
@@ -1038,7 +1041,7 @@ def test_discover_manifestation_with_size(self):
         )
         d = api.discover_state(self.EMPTY_NODESTATE)
 
-        self.assertItemsEqual(
```

Comment:
- `assertItemsEqual` actually compares the keys of dictionaries, not the contents (?!?) so the test was incorrect (i.e. comparing against a manifestation with the wrong dataset id.
- `RestartNever` compared equal to `RestartAlways` (since they both had no fields). The test was trying to say that `StartApplication` always passed `RestartNever` to the docker client, but instead just said that the policy defined above (which was `RestartAlways`) equaled `RestartNever`. The mismatch was noticed because they did in fact compare equal.